core/Tracker/VisitExcluded.php

```diff
@@ -89,8 +89,14 @@ public function isExcluded()
          * @param bool &$excluded Whether the request should be excluded or not. Initialized
          *                        to `false`. Event subscribers should set it to `true` in
          *                        order to exclude the request.
+         * @param Request $request The request object which contains all of the request's information
+         * @param bool|string $userAgent The user-agent used by the visitor
+         * @param bool|string $ip The IP-address of the visitor
+         *
+         * The additional parameters $userAgent and $ip are needed, because the values in the 
+         * object $request are empty when the constructor is called from the log-import.
```

Comment:
I don't know how the request-object is build, but look at the constructor of VisitExcluded:

```
    public function __construct(Request $request, $ip = false, $userAgent = false)
    {
        $this->spamFilter = new ReferrerSpamFilter();
        if (false === $ip) {
            $ip = $request->getIp();
        }
        if (false === $userAgent) {
            $userAgent = $request->getUserAgent();
        }
        $this->request   = $request;
        $this->idSite    = $request->getIdSite();
        $this->userAgent = $userAgent;
        $this->ip = $ip;
    }
```
You can see, that ip and user agent are taken from the given paramters when provided. So I think these values are more trustworthy than the content of the request-object.